src/framework/services/scroll/ScrollPosition.js

```diff
@@ -0,0 +1,65 @@
+
+import ThrottledEventDispatcher from '../event/ThrottledEventDispatcher';
+
+export default class ScrollPosition {
+
+  constructor() {
+    this.listeners = [];
+    this.previous = this.current = 0;
+  }
+
+  init(element) {
+    function getPosition() {
+      if (element) {
+        return element.scrollTop;
+      }
+
+      // Default to the browser scroll position. Compare between values that are
+      // used in Chrome and Firefox, respectively.
+      return Math.max(
+        document.body.scrollTop,
+        document.documentElement.scrollTop
+      );
+    }
+
+    const getDistanceFromBottom = () => {
```

Comment:
Why are you using different ways of defining `getPosition` and `getDistanceFromBottom`? I just checked the Airbnb style guide and they recommend using function declarations since they are named (https://github.com/airbnb/javascript#functions).